# origin: Debian (herbert)
# cset: n/a
# inclusion: not submitted
# description: Always call complete after unlinking a urb
# revision date: 2004-09-03

diff -urN kernel-source-2.4.26/drivers/usb/host/usb-ohci.c kernel-source-2.4.26-1/drivers/usb/host/usb-ohci.c
--- kernel-source-2.4.26/drivers/usb/host/usb-ohci.c	2004-04-14 23:05:33.000000000 +1000
+++ kernel-source-2.4.26-1/drivers/usb/host/usb-ohci.c	2004-04-17 14:24:00.000000000 +1000
@@ -861,10 +861,10 @@
 			urb->dev = NULL;
 			if (urb->transfer_flags & USB_ASYNC_UNLINK) {
 				urb->status = -ECONNRESET;
-				if (urb->complete)
-					urb->complete (urb); 
 			} else 
 				urb->status = -ENOENT;
+			if (urb->complete)
+				urb->complete (urb);
 		}	
 	} else {
 		spin_unlock_irqrestore(&ohci->ohci_lock, flags);
@@ -2173,10 +2173,10 @@
 		urb->dev = NULL;
 		if (urb->transfer_flags & USB_ASYNC_UNLINK) {
 			urb->status = -ECONNRESET;
-			if (urb->complete)
-				urb->complete (urb);
 		} else
 			urb->status = -ENOENT;
+		if (urb->complete)
+			urb->complete (urb); 
 	} else {
 		spin_unlock_irqrestore(&ohci->ohci_lock, flags);
 	}

